netbox/netbox/settings.py

```diff
@@ -63,7 +62,23 @@
 ADMINS = getattr(configuration, 'ADMINS', [])
 ALLOW_TOKEN_RETRIEVAL = getattr(configuration, 'ALLOW_TOKEN_RETRIEVAL', True)
 ALLOWED_HOSTS = getattr(configuration, 'ALLOWED_HOSTS')  # Required
-AUTH_PASSWORD_VALIDATORS = getattr(configuration, 'AUTH_PASSWORD_VALIDATORS', [])
+AUTH_PASSWORD_VALIDATORS = getattr(configuration, 'AUTH_PASSWORD_VALIDATORS', [
+    {
+        "NAME": "django.contrib.auth.password_validation.UserAttributeSimilarityValidator",
```

Comment:
This doesn't seem to validate when creating a new user. For example, I can create a user with the name & password of `NewUser123456`. However, if I then edit the user and set the same password, I get a validation error complaining that the password is too similar to the username.

AFAICT this is all stock Django functionality so I'm not sure what's going on here.

Reply:
that was tricky, needs to be in _post_clean as in clean self.instance isn't set on create.